### PR TITLE
Add topic page for unit testing, pausing/resuming time, and mocking via AsyncRead/AsyncWrite

### PR DESCRIPTION
**Pull request by barafael**
This info would have helped me early on. Let me know if deemed useful as a topic page. Perhaps it would be interesting also to have a list of useful topic pages which should be written eventually.

### CODE REVIEW

**Comment by Darksonn** [comment]
Thank you. This is definitely useful. I'll review it soon. I'm the meantime, there are some markdown code blocks that don't compile, so there is a ci failure. You should be able to click details to see why it failed.

**Comment by barafael** [comment]
At least the tests are passing now. The problem was that some code blocks were related.

**Comment by barafael** [comment]
Unfortunately, the deployment fails because

```
3:50:00 PM: Error: Unknown language: `no_run` is not registered
```

I am using no_run because the main server loop I am including in the example indeed shall not run. Not sure, maybe I should remove it anyway as it serves only to establish some context?

**Comment by Darksonn** [comment]
I'm really sorry, this feel through my cracks.

You can add a hidden "if true { return; }" at the beginning of the test. (Hide lines by starting them with #.) 

Other than getting CI to pass, this LGTM.

**Comment by barafael** [comment]
Thanks :) nice workaround!

Any topic pages that are still missing off the top of your head?

File: content/tokio/topics/testing.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+## Mocking using [`AsyncRead`] and [`AsyncWrite`]
+
+The generic traits for reading and writing asynchronously ([`AsyncRead`] and
+[`AsyncWrite`]) are implemented by, for example, sockets. They can be used for
+mocking I/O performed by a socket.
+
+Consider, for setup, this simple TCP server loop:
+
+```rust
+# if true { return }
```
Comment (Darksonn) [inline content/tokio/topics/testing.md:77]: ... this is outside the main function. How can this work? There must be something wrong with our CI setup.

Comment (barafael) [reply content/tokio/topics/testing.md]: ... it's been a long day?! fixed

**Review by Darksonn — APPROVED** [review]
Thanks.

**Comment by Darksonn** [comment]
> Any topic pages that are still missing off the top of your head?

Getting started with Axum comes to mind. @davidpdrsn thoughts?

**Comment by davidpdrsn** [comment]
Yeah a getting started guide for axum would be great. I don't have capacity to write something but I can help review. There is also some relevant discussion here https://github.com/tokio-rs/axum/issues/1811.

**Pull request merged**